src/Opulence/Ioc/Container.php

```diff
@@ -146,15 +146,19 @@ public function resolve(string $interface)
 
         switch (get_class($binding)) {
             case InstanceBinding::class:
+                /** @var InstanceBinding $binding */
                 return $binding->getInstance();
             case ClassBinding::class:
+                /** @var ClassBinding $binding */
                 $instance = $this->resolveClass(
                     $binding->getConcreteClass(),
                     $binding->getConstructorPrimitives()
                 );
                 break;
             case FactoryBinding::class:
-                $instance = ($binding->getFactory())();
+                /** @var FactoryBinding $binding */
+                $factory = $binding->getFactory();
+                $instance = $factory();
```

Comment:
I agree that's the most succinct code, but my IDE is (incorrectly) giving me warnings about it.